src/firefly/js/visualize/ui/MatchLockDropDown.jsx

```diff
@@ -6,14 +6,44 @@ import {PlotAttribute} from '../PlotAttribute';
 import {SingleColumnMenu} from '../../ui/DropDownMenu.jsx';
 import {DropDownVerticalSeparator, ToolbarButton} from '../../ui/ToolbarButton.jsx';
 import {DropDownToolbarButton} from '../../ui/DropDownToolbarButton.jsx';
-
 import MATCH_LOCKED from 'html/images/28x28_Match_Locked.png';
 import MATCH_UNLOCKED from 'html/images/28x28_Match_Unlocked.png';
 import {isImage} from '../WebPlot';
+import {showInfoPopup} from 'firefly/ui/PopupUtil';
 
 function changeMatchType(vr, matchType, lockMatch) {
     const plot= primePlot(vr);
     if (!plot) return;
+    //check for images without WCS when aligning by WCS or Target, and display appropriate warning
+    if(matchType.key === 'Standard' || matchType.key === 'Target') {
+        const plotViewObjs = getPlotViewAry(vr);
+        const warningTitles = [];
+        for (const i in plotViewObjs) {
+            const currPlots = plotViewObjs[i].plots;
+            for (const j in currPlots) {
+                const plt = currPlots[j];
+                const checkWCSExists = hasWCSProjection(plt);
+                if (!checkWCSExists) {
+                    if (!warningTitles.includes(plt.title)) {
+                        warningTitles.push(plt.title);
+                    }
+                }
+            }
+        }
```

Comment:
`for-in` loop is better suited for object.  For array, I would try `forEach`.  i.e.
```
getPlotViewAry(vr)?.forEach( (currPlots) => {
	currPlots.forEach( (plt) => {
		const checkWCSExists = hasWCSProjection(plt);
		if (!checkWCSExists) {
			if (!warningTitles.includes(plt.title)) {
				warningTitles.push(plt.title);
			}
		}
	});
});
```

Reply:
Got it, that makes sense. I think also by adding obj in the name I got myself confused and thought of it as an object, although it was always an array. Made the change.